src/steps/findings/index.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  createDirectRelationship,
+  IntegrationStepExecutionContext,
+} from '@jupiterone/integration-sdk-core';
+import { CiscoAmpVulnerability, createServicesClient } from '../../collector';
+import { IntegrationConfig } from '../../config';
+import { Entities, Relationships, Steps } from '../constants';
+import { createFindingEntity, createVulnerabilityEntity } from './converter';
+
+export async function fetchVulnerabilities({
+  instance,
+  jobState,
+  logger,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const client = createServicesClient(instance);
+
+  await client.iterateVulnerabilities(async (vuln: CiscoAmpVulnerability) => {
+    for (const cve of vuln.cves) {
+      // skip entity creation if we have encountered the cve before
+      let vulnEntity = await jobState.findEntity(cve.id.toLowerCase());
+      if (!vulnEntity) {
+        vulnEntity = await jobState.addEntity(createVulnerabilityEntity(cve));
+      }
+
+      // When there are over 1000 computers, only the first 1000 will appear on
+      // the vulnerability response. This case should be handled by
+      // using IterateComputersWithVulnerability
```

Comment:
Are we handling this case of > 1000 computers anywhere using the `IterateComputersWithVulnerability` function?

Reply:
Not currently. I have the `IterateComputersWithVulnerability` function created in the client. When I started to add handling for this case it started to make this PR much bigger and riskier than I'd like. There are already many changes from this and the last two PRs that haven't been deployed. Also, testing this case isn't trivial because we don't have enough data to do it with a recording.

To try to reduce risk, getting this deployed will help being much more confidence when adding handling for this special case.

Reply:
Sounds good!  Just wanted to make sure I understood what was going on with it and what the plan was.  Thanks!